### ISSUE TITLE
difference sampling (multistock / oldreagents) broken

### ISSUE THREAD

**Issue by ipendlet** [darkreactions/ESCALATE_Capture#96]
I think this is going to be a rather complex debug.  First, whoever is programming this function should really wrap their head around what the goal of this feature actually is:  we are attempting to use old inputs (a previous xlsx) as the "old" reagent, or the "old stock solution" and remove the regions that these old reagents were able to access from the total region of the available concentration space.   

For example, in the case of a simple iodide perovskite experiment we can target the grey region as the full space.  The blue region is what has already been sampled (old sampling, poor).  The yellow region is the space that is accessible for all of the PbI2 containing compounds, while the yellow space highlighted by the black triangle is the region accessible by only 2 of the stock solutions.  

For examples of the failings of the current multistock sampling see issue 
A demonstration of a run accessing a related "yellow" region.  LBL_Multistock_Edited.xlsx 

The difference sampling is clearly not working either:
![search space](https://user-images.githubusercontent.com/22350032/69741003-47c93280-1108-11ea-8640-0886aca092f9.png)
  

A few challenges

**Comment by jschrier** [comment]
Yeah, I don't think this is actually a true bug, but rather an input problem. [Although I can't really make sense of how to read the spreadsheets in issue 96] 

As commented to IP on Slack DM 26 Nov 2019—this appears to be a misunderstanding of how the difference sampler works.  The "new" region is defined by a set of reagents (which may or many not be the same), and the "old/excluded" region should also be defined with all the relevant reagents (this includes neat GBL and neat FAH).  

I'll remind you of the usage demo from back on 31 Aug 2019... https://www.wolframcloud.com/env/jschrier0/Published/2019.08.31_new_region_sampler.nb

[FWIW, the updated code for sampling experiments containing >3 solute species also speeds this up by 10x as a "side effect"]

**Comment by ipendlet** [comment]
If X=pbi2 conc, and y=amine, the new region explores the region of space in low y, under the original pbi2 solution line.   If we think of the new region as being everything under the old region (in terms of y), than in order to access that new  space we must include the old pbi2 solution and omit the amine solution -- that's it. We don't actually need any extra (difference) sampling tools, just select the stock solutions appropriately.

Explanation/demonstration of correctness: https://www.wolframcloud.com/obj/jschrier0/Published/2019.11.27_do_we_actually_need_difference_sampling.nb

ifference sampler:  Consider the following case of 1,4-DAB.  It would be useful to sample the entire space, less the convex hull of experiments that have already been done...
[The wrinkle, of course, is that it is not the difference defined by solution points, but by a more arbitrary geometric object]
No specific action suggested, but something to keep in mind.

**Comment by ipendlet** [comment]
<img width="427" alt="Screen Shot 2019-12-10 at 11 53 31 AM" src="https://user-images.githubusercontent.com/22350032/70568632-449d6000-1b66-11ea-9193-27a29cc0ab34.png">
